platform_unix.cpp

```diff
@@ -51,6 +51,11 @@ char *show_file_picker() {
 					"_Open", GTK_RESPONSE_ACCEPT,
 					NULL );
 
+	// Filter file types
+	GtkFileFilter *filter = gtk_file_filter_new();
+	gtk_file_filter_add_pattern(filter, "*.brd");
+	gtk_file_chooser_add_filter(dialog, filter);
```

Comment:
This needs to be `gtk_file_chooser_add_filter(GTK_FILE_CHOOSER(dialog), filter);` - turns out the dialog needs casting to the `GtkFileChooser` interface to use `gtk_file_chooser_add_filter` on it.